src/components/DragList/DragListItem.js

```diff
@@ -0,0 +1,150 @@
+ */
+class DragListItem extends React.PureComponent {
+  static Consumer = DragContext.Consumer;
+
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    /**
+     * DragList.Items that interact with each other should share the same type string value.
+     */
+    itemType: PropTypes.string.isRequired,
+    /**
+     * Index of the item. Used for moving items around
+     */
+    index: PropTypes.number.isRequired,
+    /**
+     * Called with (fromIndex, toIndex) when the item is moved to a new position
+     */
+    onMove: PropTypes.func.isRequired,
+  };
+
+  componentDidMount() {
+    // Use empty image as a drag preview so browsers don't draw it
+    // and we can draw whatever we want on the custom drag layer instead.
+    this.props.connectDragPreview(getEmptyImage(), {
+      // IE fallback: specify that we'd rather screenshot the node
+      // when it already knows it's being dragged so we can hide it with CSS.
+      captureDraggingState: true,
+    });
+  }
+
+  wrapDragHandle = dragHandleNode =>
+    /**
+     * The element passed to connectDragSource must be a plain
+     * React element, not a component.
+     */
+    this.props.connectDragSource(
+      <div className="DragListItem--handle">{dragHandleNode}</div>,
+    );
+
+  render() {
+    const {
+      props: {
+        connectDragSource,
+        connectDragPreview,
+        connectDropTarget,
+        isDragging,
+        onMove,
+        index,
+        children,
+      },
+      wrapDragHandle,
+    } = this;
+
+    return connectDropTarget(
+      <div>
+        <DragContext.Provider value={{ isDragging, wrapDragHandle }}>
+          {children}
+        </DragContext.Provider>
+      </div>,
+    );
+  }
+}
+
+const dragSource = {
+  beginDrag(props, monitor, component) {
+    const domNode = ReactDOM.findDOMNode(component);
+    const clientRect = domNode.getBoundingClientRect();
+    return {
+      index: props.index,
+      children: props.children,
+      dimensions: clientRect,
+    };
+  },
+};
+
+const dropTarget = {
+  hover(props, monitor, component) {
+    const dragIndex = monitor.getItem().index;
+    const hoverIndex = props.index;
+
+    // Don't replace items with themselves
+    if (dragIndex === hoverIndex) {
+      return;
+    }
+
+    // Determine rectangle on screen
+    const hoverBoundingRect = ReactDOM.findDOMNode(
+      component,
+    ).getBoundingClientRect();
+
+    // Get vertical middle
+    const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+    // Determine mouse position
+    const clientOffset = monitor.getClientOffset();
+
+    // Get pixels to the top
+    const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+    // Only perform the move when the mouse has crossed half of the items height
+    // When dragging downwards, only move when the cursor is below 50%
+    // When dragging upwards, only move when the cursor is above 50%
```

Comment:
FWIW, I found that I had to move much further than I felt I should in order to reorder items when playing around with this. Intuitively I felt that the drop target should be whichever space my cursor was currently hovering over, but in practice I had to move over half the distance further across the element before the target changed (as specified here). Just the fact that I had to move so far, but then I could move backward and nothing would change, felt very non-physical to me.

Based on [this article](https://medium.com/@alexandereardon/rethinking-drag-and-drop-d9f5770b4e6b) I feel like the best solution is to move items when the center point of the item itself has crossed the space boundary. Since our cursor is already centered on the item, that would be when the cursor crosses that boundary. We could probably make this feel more natural by animating the position of other items if possible.

Reply:
I think the behavior change is sufficient.